### PR TITLE
Feature/angular2 jsonapi

### PR DESCRIPTION
**Pull request by mrjmd**
Work in progress, it is close but does not work currently.

### CODE REVIEW

File: src/app/components/recipe-details/recipe-details.component.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <md-card *ngIf="recipe">
   <div>
-    <span id="title">{{recipe.data.attributes.title}}</span>
-    <span id="difficulty">{{recipe.data.attributes.difficulty}}</span>
+    <span id="title">{{recipe.title}}</span>
```
Comment (joaogarin) [inline src/app/components/recipe-details/recipe-details.component.html]: :+1:  much better!

File: src/app/models/file.model.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  @Attribute()
+  size: number;
+
+  @Attribute()
+  updatedAt: Date;
+
+  @Attribute()
+  createdAt: Date;
+
+}
```
Comment (joaogarin) [inline src/app/models/file.model.ts]: end line in the bottom

File: src/app/models/image.model.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  @Attribute()
+  updatedAt: Date;
+
+  @BelongsTo()
+  imageFile: File;
+
+  @BelongsTo()
+  thumbnail: File;
+
+}
```
Comment (joaogarin) [inline src/app/models/image.model.ts]: end line in the bottom

File: src/app/services/backend.service.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import { Http } from '@angular/http';
```
Comment (mrjmd) [inline src/app/services/backend.service.ts]: This file is where I can feel the technical debt / shoddy patchwork code accumulating. It needs a proper scrubbing, but for this iteration, it works and moves us forward.

File: src/app/models/file.model.ts (hunk 2 above)
Comment (mrjmd) [inline src/app/models/file.model.ts]: fixed

File: src/app/models/image.model.ts (hunk 3 above)
Comment (mrjmd) [inline src/app/models/image.model.ts]: fixed

File: src/app/components/recipe/recipe.component.spec.ts [hunk 5]
```diff
@@ -7,6 +8,14 @@ import { MdCardModule, MdButtonModule, MdChipsModule, MdIconModule } from '@angu
 import { RecipeComponent } from './recipe.component';
 import { CardComponent } from './../card/card.component';
 
+const RECIPE_DATA = {
+    id: '1',
+    attributes: {
+        title: 'Recipe title',
+    }
+};
+const recipe: Recipe = new Recipe(null, RECIPE_DATA);
```
Comment (joaogarin) [inline src/app/components/recipe/recipe.component.spec.ts]: Interesting that this is failing on travis but working for me locally. 

(https://github.com/ghidoz/angular2-jsonapi/issues/79)

**Pull request merged**